containers/test-apps/jwks-server/src/jwks_server/handler.clj

```diff
@@ -0,0 +1,112 @@
+;;
+;; Copyright (c) Two Sigma Open Source, LLC
+;;
+;; Licensed under the Apache License, Version 2.0 (the "License");
+;; you may not use this file except in compliance with the License.
+;; You may obtain a copy of the License at
+;;
+;;  http://www.apache.org/licenses/LICENSE-2.0
+;;
+;; Unless required by applicable law or agreed to in writing, software
+;; distributed under the License is distributed on an "AS IS" BASIS,
+;; WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+;; See the License for the specific language governing permissions and
+;; limitations under the License.
+;;
+(ns jwks-server.handler
+  (:require [buddy.core.keys :as buddy-keys]
+            [buddy.sign.jwt :as jwt]
+            [clojure.data.json :as json]
+            [clojure.tools.logging :as log]
+            [clojure.string :as str]
+            [jwks-server.config :as config]
+            [plumbing.core :as pc]))
+
+(defn prepare-response
+  "Prepares and returns a standard response"
+  ([{:keys [query-string request-method uri]} message status]
+   (prepare-response (cond-> {"message" message
+                              "request-method" request-method
+                              "uri" uri}
+                       (not (str/blank? query-string))
+                       (assoc "query-string" query-string))
+                     status))
+  ([data-map status]
+   {:body (json/write-str data-map)
+    :headers {"content-type" "application/json"
+              "server" "jwks-server"}
+    :status status}))
+
+(defn generate-jwt-access-token
+  "Generates the JWT access token using the provided private key."
+  [alg jwk-entry payload header]
+  (let [private-key (-> jwk-entry pc/keywordize-map buddy-keys/jwk->private-key)
+        options {:alg alg :header header}]
+    (jwt/sign payload private-key options)))
+
+(defn process-get-token-request
+  "Retrieves an JWT access token generated using a random EdDSA key."
+  [{:keys [query-string]}]
+  (log/info "query string:" query-string)
+  (let [{:strs [host]} (try
+                         (when-not (str/blank? query-string)
+                           (->> (str/split query-string #"&")
```

Comment:
We should use ring's built in param parsing: https://github.com/ring-clojure/ring/blob/master/ring-core/src/ring/middleware/params.clj

Reply:
URL parameters can be URL encoded, which wasn't being handled.